pkg/kv/store.go

```diff
@@ -125,6 +125,12 @@ type EntriesIterator interface {
 	Close()
 }
 
+type ResultIterator interface {
```

Comment:
Document the iterator which is now part of the kv interface. Honestly, not loving the name. `Result` is too general and `Entry` in the context of KV makes more sense to me. I realized how we ended up here, since we needed another name for an iterator, but if I understand correctly it returns the `Value`, or `ValueWithPredicate`, no? So maybe name it `ValueIterator`?

Reply:
Reading it again, the interface doesn't return `EntriesIterator` anywhere, so I don't it should be named at all. WDYT?